### PR TITLE
Cleanup PK mod

### PR DESCRIPTION
**Pull request by mugling**
Similar to `blazemod` the PK mod is badly formatted and contains many errors. Recently (currently?) it fails to load without `debugmsg` and is difficult to keep in sync when adding new features.

This PR will extend the linter to cover all of `pk_item_*`. This should fix both existing errors and hopefully prevent new ones being introduced.

Required extensions to the linter are as usual backed by further unit tests in `tools/format/examples`. Most of the line count (regarding the linter) is due to adding support for `countdown_action`.

Error corrections to the mod are made in a separate commit to formatting and another developer or @pisskop needs to review those manually.

### CODE REVIEW

**Comment by mugling** [comment]
Ready

File: data/mods/PKs_Rebalance/items/gun.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 [
-
-    {
+  {
     "id": "pipe_launcher40mm",
```
Comment (pisskop) [inline data/mods/PKs_Rebalance/items/gun.json]: This is moved over to the file 'override'.

Comment (mugling) [inline data/mods/PKs_Rebalance/items/gun.json]: Which one?

**Comment by pisskop** [comment]
Sorry.  the 40mm tube launcher was moved from item_guns to item_override because it fit better.


I dont necessarily have a problem with reformatting what  I have, which seems the be the vast majority of what this is, but Im not quite sure I understand what longer-term implications this PR is supposed to have?

The changes in format I see are the removal of the 4 spaces before each line and separating the close brackets of the old id and the open bracket of the new one.

I started most of these item/recipes/types by copying an example in the json.  Most of these are directly copied from the json examples in recent months, and some, such as the entire 'override' item file is as recent as last week.

**Comment by mugling** [comment]
> I dont necessarily have a problem with reformatting what I have, which seems the be the vast majority of what this is

Formatting is helpful. A lot of the errors (see 7467795) are the result of misleading indentation.

> but Im not quite sure I understand what longer-term implications this PR is supposed to have?

If either you use an undefined field or another developer changes the JSON format and fails to update your mod then the buildbot will complain. This should help to keep your mod in sync with `core`.

**Comment by pisskop** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/pull/18203
Small tweaks and misc changes, from 7 days ago

https://github.com/CleverRaven/Cataclysm-DDA/pull/18219
Wildy items and mechanical things, 3 days ago

https://github.com/CleverRaven/Cataclysm-DDA/pull/18277
Item overrides, 1 day ago

**Comment by mugling** [comment]
@pisskop all of those PR's are already merged?

**Comment by pisskop** [comment]
Yes.  no outstanding prs

**Comment by pisskop** [comment]
Well, if this will be an extended affair Ill go and fix up the hard errors in a seperate pr; it seems that the override clause is no longer needed and its throwing errors

**Comment by mugling** [comment]
> Well, if this will be an extended affair Ill go and fix up the hard errors in a seperate pr; it seems that the override clause is no longer needed and its throwing errors

That will turn this PR in to a merge conflict. Check you are happy with the changes made in 4449a4c and we can continue from there.

**Comment by pisskop** [comment]
yea, its fine.

**Comment by mugling** [comment]
What platform do you compile on @pisskop?

**Comment by pisskop** [comment]
cmake?

I havent progressed enough down DIY C++ to really know

**Comment by mugling** [comment]
Sorry to be clear, linux (or a variant), windows or mac?

**Comment by pisskop** [comment]
Linux

**Comment by mugling** [comment]
> Linux

Once linting is enabled for your mod you need to do `make json-lint` before pushing your changes. It will let you know if there are any problems.

I'll probably get round to adding a makefile target that automatically fixes any problems (à la `make astyle`)

**Comment by mugling** [comment]
> yea, its fine.

If you change any of the files under `items` you should run the linter via `./tools/cleanup.sh none data/mods/PK_Rebalance/items/*`.

This will ensure any future changes you make are valid and will also cause any PR that fails to update your mod to be rejected by the buildbot.

**Comment by Rivet-the-Zombie** [comment]
@mugling, please don't merge your own PRs. We always let another dev merge our own stuff.

**Comment by mugling** [comment]
@Rivet-the-Zombie I'm taking it from #18298 that automated changes via the linter were acceptable independent of size:

> If one other dev agrees (or at least "agrees if someone else agrees"), then I also agree. Then we'll have the active dev majority.

> So long as everything is properly tested, I see no problem with this PR.

**Comment by Rivet-the-Zombie** [comment]
I must have misunderstood you; I meant that I was fine with merging large PRs if they've been properly tested.

**Comment by mugling** [comment]
> I must have misunderstood you; I meant that I was fine with merging large PRs if they've been properly tested.

No worries, I've set out more clearly what I have in mind in #18330 for the you and the other developers to weigh in their opinions

**Pull request merged**